src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1751,10 +1680,18 @@ static void recv_message_ready(void* arg, grpc_error* error) {
       static_cast<subchannel_call_retry_state*>(
           grpc_connected_subchannel_call_get_parent_data(
               batch_data->subchannel_call));
+  ++retry_state->completed_recv_message_count;
+  // If a retry was already dispatched, then we're not going to use the
+  // result of this recv_message op, so do nothing.
+  if (retry_state->retry_dispatched) {
```

Comment:
This statement and comment look like a near-duplicate of line 1597. I think this is indicative that the code needs a refactor. We should consider this in a later PR.